src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx

```diff
@@ -0,0 +1,291 @@
+import React from 'react';
+import * as ReactRouter from 'react-router';
+import * as Sentry from '@sentry/browser';
+
+import {Client} from 'app/api';
+import {Group, Organization, Project} from 'app/types';
+import {IconInfo} from 'app/icons';
+import {PageContent} from 'app/styles/organization';
+import {callIfFunction} from 'app/utils/callIfFunction';
+import {t} from 'app/locale';
+import Alert from 'app/components/alert';
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import GroupStore from 'app/stores/groupStore';
+import LoadingError from 'app/components/loadingError';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import Projects from 'app/utils/projects';
+import SentryTypes from 'app/sentryTypes';
+import recreateRoute from 'app/utils/recreateRoute';
+import withApi from 'app/utils/withApi';
+import withProfiler from 'app/utils/withProfiler';
+
+import {ERROR_TYPES} from './constants';
+import GroupHeader from './header';
+
+type Error = typeof ERROR_TYPES[keyof typeof ERROR_TYPES] | null;
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  environments: string[];
+  children: React.ReactNode;
+  finishProfile: () => void;
+} & ReactRouter.RouteComponentProps<{orgId: string; groupId: string}, {}>;
+
+type State = {
+  group: Group | null;
+  loading: boolean;
+  error: boolean;
+  errorType: Error;
+  project: null | Pick<Project, 'platform' | 'id' | 'slug'>;
+};
+
+class GroupDetails extends React.Component<Props, State> {
+  static childContextTypes = {
+    group: SentryTypes.Group,
+    location: PropTypes.object,
+  };
+
+  state = this.initialState;
+
+  getChildContext() {
+    return {
+      group: this.state.group,
+      location: this.props.location,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(_prevProps, prevState: State) {
+    if (prevState.loading && !this.state.loading) {
+      callIfFunction(this.props.finishProfile);
+    }
+  }
+
+  componentWillUnmount() {
+    callIfFunction(this.listener);
+  }
+
+  get initialState(): State {
+    return {
+      group: null,
+      loading: true,
+      error: false,
+      errorType: null,
+      project: null,
+    };
+  }
+
+  remountComponent() {
+    this.setState(this.initialState);
+  }
+
+  get groupDetailsEndpoint() {
+    return `/issues/${this.props.params.groupId}/`;
+  }
+
+  async fetchData() {
+    const {environments, api} = this.props;
+
+    try {
+      const data = await api.requestPromise(this.groupDetailsEndpoint, {
+        query: {
+          // Note, we do not want to include the environment key at all if there are no environments
+          ...(environments ? {environment: environments} : {}),
+        },
+      });
+
+      // TODO(billy): See if this is even in use and if not, can we just rip this out?
+      if (this.props.params.groupId !== data.id) {
+        const {routes, params, location} = this.props;
```

Comment:
Should we have a metric for this to see if it gets hit in the wild?

Reply:
👍 Yeah that's the intention